src/ast/passes/codegen_llvm.cpp

```diff
@@ -151,16 +151,98 @@
                                 StackType stack_type,
                                 const location &loc)
 {
-  Value *stackid = b_.CreateGetStackId(
-      ctx_, ident == "ustack", stack_type, loc);
+  if (!murmur_hash_2_func_)
+    murmur_hash_2_func_ = createMurmurHash2Func();
+
+  bool is_ustack = ident == "ustack";
+  auto uint64_size = sizeof(uint64_t);
+
+  AllocaInst *stackid = b_.CreateAllocaBPF(CreateUInt64(), "stackid");
+
+  Function *parent = b_.GetInsertBlock()->getParent();
+  BasicBlock *stack_scratch_failure = BasicBlock::Create(
+      module_->getContext(), "stack_scratch_failure", parent);
+  BasicBlock *merge_block = BasicBlock::Create(module_->getContext(),
+                                               "merge_block",
+                                               parent);
+
+  Value *stack_trace = b_.CreateGetStackScratchMap(stack_type,
+                                                   stack_scratch_failure,
+                                                   loc);
+  b_.CREATE_MEMSET(
+      stack_trace, b_.getInt8(0), uint64_size * stack_type.limit, 1);
+
+  BasicBlock *get_stack_success = BasicBlock::Create(module_->getContext(),
+                                                     "get_stack_success",
+                                                     parent);
+  BasicBlock *get_stack_fail = BasicBlock::Create(module_->getContext(),
+                                                  "get_stack_fail",
+                                                  parent);
+
+  Value *ptr_to_stack_trace = b_.CreatePointerCast(stack_trace,
+                                                   b_.GET_PTR_TY());
+  Value *stack_size = b_.CreateGetStack(
+      ctx_, is_ustack, ptr_to_stack_trace, stack_type, loc);
+  Value *condition = b_.CreateICmpSGE(stack_size, b_.getInt32(uint64_size));
```

Comment:
Javier mentioned <2 frames is usually an error. But maybe we shouldn't
codify that? Not sure.

Reply:
Yeah not sure either. Maybe just leave it.